releasenotes/notes/add-bit-register-rust-979dd113aefb3563.yaml

```diff
@@ -0,0 +1,23 @@
+---
+features_circuits:
+  - |
+   Python classes :class:`.Bit` and :class:`.Register` have been ported to Rust
+   along with all their respective subclasses including:
+   
+   * :class:`.Qubit` and :class:`.QuantumRegister`.
+   * :class:`.Clbit` and :class:`.ClassicalRegister`.
+   * :class:`.AncillaQubit` and :class:`.AncillaRegister`.
+upgrade_circuits:
+  - |
+    :class:`.DAGCircuit` and :class:`.CircuitData` will now store :class:`.Qubit`,
+    :class:`.Clbit`, as well as its register counterparts, in Rust space.
+  - The :attr:`.QuantumCircuit._qubit_indices` and :attr:`.QuantumCircuit._clbit_indices`
+    attributes of the :class:`.QuantumCircuit` class have been moved to live in 
+    rust space and store rust native data.
+  - The :meth:`.QuantumCircuit._qbit_argument_conversion` and :meth:`.QuantumCircuit._cbit_argument_conversion`
+    have been moved to Rust to operate with native data.
```

Comment:
What are the user implications for this? An upgrade note is a potential call for action from end users saying you may need to change something when you upgrade. Just operating in rust space isn't really a user facing change. Looking at the test changes it's the note below about the `is` checks, maybe the `TypeError` instead of `DAGCircuitError`.

Reply:
Hmm, that's right, these functions weren't exposed to the public either way, so I could remove those parts.

Reply:
Done in 0e669440a.